db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ProjectRestrictNode.java

```diff
@@ -1719,6 +1721,10 @@ private void generateMinion(ExpressionClassBuilder acb,
             // which is the static field that "points" to the userExprFun
             // that evaluates the where clause.
             acb.pushMethodReference(mb,userExprFun);
+
+            CollectNodesVisitor cnv = new CollectNodesVisitor(ParameterNode.class);
+            constantRestriction.accept(cnv);
+            mb.push(!cnv.getList().isEmpty());
```

Comment:
Since you are not using the collected nodes, it would be more efficient to have a mode where CollectNodesVisitor stops searching once the first node is found.  Maybe create a new constructor that accepts a boolean as a 2nd parameter.

Reply:
You're right. I just remember that we have a `HasNodeVisitor`. Will use that instead.